mlrun/db/httpdb.py

```diff
@@ -1461,9 +1462,10 @@ def submit_pipeline(
                 cleanup_ttl=cleanup_ttl,
                 op_transformers=ops,
             )
-            kfp.compiler.Compiler().compile(
-                pipeline, pipe_file, type_check=False, pipeline_conf=conf
-            )
+            with unsupported.disable_unsupported_external_features():
```

Comment:
checkout `min_nuclio_versions` functionality as well, this is how we restrict certain nuclio versions according to the minimal version being supported
perhaps can unite the logic under same idea

Reply:
I see two main opportunities here: 

**(1) How to detect unsupported features** 
The Nuclio function seems to get its data from MLConf. I don't think we can do that to detect if `ParallelFor` is going to be run by the user. We could inspect their source code directly though.

**(2) How to wrap code that needs to be checked** 
A decorator-style wrapper would work indeed, but it won't really look like a decorator in this case, because we want to wrap the user-defined function (whose pointer is held by the `pipeline` variable). So, it'll work, but it'll look more like `pipeline = disable_unsuported_features(pipeline)` than a true decorator. If that's preferable, let me know. I went with the context manager approach because the idea of making changes to an environment before **and** after something runs felt very context-like to me.

I'm happy to brainstorm some ideas with you. Wanna do a quick PS session together?